integration/client_delete_serviceInstance_test.go

```diff
@@ -0,0 +1,204 @@
+					if condition.Type == "Ready" && condition.Status != "False" {
+						t.Fatalf("Expected ready condition to be false and got %v", condition.Status)
+
+					}
+				}
+
+				pods := &v1.PodList{}
+				if err = getResource("pods", pods); err != nil {
+					t.Fatalf("unexpected error %v", err)
+
+				}
+				for _, pod := range pods.Items {
+					if strings.Contains(pod.ObjectMeta.Name, "fh-sync-server") {
+						for _, condition := range pod.Status.Conditions {
+							if condition.Type == "Ready" && condition.Status != "False" {
+								t.Fatalf("Expected fh-sync-server pod (%v) ready condition to be false and got %v", pod.ObjectMeta.Name, condition.Status)
+							}
+						}
+					}
+				}
+
+				deployments := &appsv1.DeploymentList{}
+				if err = getResource("deployments", deployments); err != nil {
+					t.Fatalf("unexpected error %v", err)
+
+				}
+				for _, dp := range deployments.Items {
+					if strings.Contains(dp.ObjectMeta.Name, "fh-sync-server") {
+						for _, condition := range dp.Status.Conditions {
+							if condition.Type == "Available" && condition.Status != "False" {
+								t.Fatalf("Expected fh-sync-server deployment (%v) available condition to be false and got %v", dp.ObjectMeta.Name, condition.Status)
+							}
+						}
+					}
+				}
+
+				// routes := &extv1.IngressList{}
+				// if err = getResource("routes", routes); err != nil {
+				// 	t.Fatalf("unexpected error %v", err)
+
+				// }
+				// for _, route := range routes.Items {
+				// 	if strings.Contains(route.ObjectMeta.Name, "fh-sync-server") {
+				// 		for _, condition := range route.Status {
+				// 			if condition.Type == "Available" && condition.Status != "False" {
+				// 				t.Fatalf("Expected fh-sync-server deployment (%v) available condition to be false and got %v", route.ObjectMeta.Name, condition.Status)
+				// 			}
+				// 		}
+				// 	}
+				// 	//fmt.Println(dp)
+				// }
+
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			cmd := exec.Command(*executable, test.args...)
+			output, err := cmd.CombinedOutput()
+			if err != nil && test.expectedError == nil {
+				t.Fatalf("Failed to create binding: '%s', with error: '%s'", output, err.Error())
+			}
+			if err == nil && test.expectedError != nil {
+				t.Fatalf("expected error: '%s', got: nil, output: '%s'", test.expectedError.Error(), output)
+			}
+			if err != nil && err.Error() != test.expectedError.Error() {
+				t.Fatalf("expected error: '%s', got: '%s', output: '%s'", test.expectedError.Error(), err.Error(), output)
+			}
+			if test.fixture != "" {
+				actual := strings.TrimSpace(string(output))
+				expected := strings.TrimSpace(LoadSnapshot(t, deleteServicetestPath+test.fixture))
+				if actual != expected {
+					t.Fatalf("actual = \n'%s', expected = \n'%s'", actual, expected)
+				}
+			}
+
+			if test.validate != nil {
+				test.validate(t)
+			}
+		})
+	}
+}
+
+func createInstance(t *testing.T, si *ProvisionServiceParams) {
```

Comment:
we should put our common methods into a shared utils file

Reply:
While this is not a bad idea, I would prefer to avoid utils files. If the tests are in the same package then it doesn't really matter where they are declared as the tests will have access to them at the package level.

Reply:
@maleck13 ok, in that case we'll just export them